### PR TITLE
[CSPM] Add a system-probe module to perform compliance checks

### PR DESCRIPTION
**Pull request by jinroh**
### What does this PR do?

This PR adds a new module to the system-probe binary in order to perform compliance checks accross containers. Specifically to fetch containerized DB applications configurations on which we can perform compliance checks.

The module exposes an endpoint that loads and parses the configuration files of a few known databases applications: Cassandra, MondoDB and PostgreSQL. It then returns the parsed data to the caller, security-agent in our case. The configuration file is located either by parsing the command line arguments of the associated DB process (identified by its PID) or looking at default known paths in /etc.

### Motivation

Being able to perform cross-container compliance checks from security-agent relying on system-probe.

### Additional Notes

Performance impact:

The overall overhead of this module should be minimal. Also it will not be called by default in the release to have time to QA the impact.

1. adding dependency on pkg/compliance/dbconfig only adds ~63KB to the unstripped binary size
2. compliance module's endpoint should only be called one every three hours by our security-agent.
3. the actual workload of this module only consists of loading a few configuration files of typically ~10s of kilobytes and perform a basic parsing to structure the data in JSON. Guardrails are in place to make sure we do not read more that 1MB of data per config files.

### Describe how to test/QA your changes

Launching the agent in a k8s environment and setting `compliance_config.db_benchmarks.enabled` to `true` alongside a container running a postgres, cassandra or mongodb and see if configuration logs are sent to the compliance track.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by sgnn7 — APPROVED** [review]
LGTM for ASC-owned `config.go`

**Review by 0intro — APPROVED** [review]

**Comment by 0intro** [comment]
I think it may be worth adding release notes, since it's a new feature (even if it's disabled by default).

**Review by lebauce — APPROVED** [review]

**Review by brycekahle — APPROVED** [review]

File: cmd/system-probe/modules/compliance.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package modules
```
Comment (brycekahle) [inline cmd/system-probe/modules/compliance.go:6]: I think you might want a build tag `//go:build linux` here?

Comment (brycekahle) [reply cmd/system-probe/modules/compliance.go:6]: I just noticed the increase in imports for windows system-probe

Comment (brycekahle) [reply cmd/system-probe/modules/compliance.go:6]: `compliance_test.go` too

Comment (jinroh) [reply cmd/system-probe/modules/compliance.go:8]: ce919296b1343351f7c75728d5f4896a3892b0ed

**Review by brycekahle — APPROVED** [review]

**Comment by jinroh** [comment]
Failures on `Windows unit tests / windows-unit-tests (pull_request)` are unrelated and fixed on main.

**Pull request merged**